contentcuration/contentcuration/viewsets/user.py

```diff
@@ -266,7 +267,30 @@ def create_from_changes(self, changes):
 
     def delete_from_changes(self, changes):
         return self._handle_relationship_changes(changes)
-
+    @action(detail=True, methods=['delete'])
+    def remove_self(self, request, pk=None):
+        """
+        Allows a user to remove themselves from a channel as a viewer.
+        """
+        user = self.get_object()
+        channel_id = request.query_params.get('channel_id', None)
+
+        if not channel_id:
+            return Response({'detail': 'channel_id is required.'}, status=status.HTTP_400_BAD_REQUEST)
+
+
+        channel = Channel.objects.get(id=channel_id)
+        if not channel:
+            return Response({'detail': 'Channel not found.'}, status=status.HTTP_404_NOT_FOUND)
+
+        if request.user != user and not request.user.can_edit(channel_id):
+            return Response({'detail': 'You do not have permission to remove this user.'}, status=status.HTTP_403_FORBIDDEN)
```

Comment:
Lets return HttpResponseForbidden. You can specify "You do not have permission to remove this user." as the message, and maybe the id of the user too?